gspread/utils.py

```diff
@@ -979,6 +985,99 @@ def to_records(
     return [dict(zip(headers, row)) for row in values]
 
 
+def _expand_right(values: List[List[str]], start: int, end: int, row: int) -> int:
+    # find rightmost value
+    for column in range(start, end):
+        if values[row][column] == "":
+            return column
+
+    return end
+
+
+def _expand_bottom(values: List[List[str]], start: int, end: int, col: int) -> int:
+    # find the bottommost value
+    for rows in range(start, end):
+        print("bottom check row: {} col {}".format(rows, col))
+        if values[rows][col] == "":
+            return rows
+
+    return end
+
+
+def find_table(
+    values: List[List[str]], direction: TableDirection, start_range: str
+) -> List[List[str]]:
+    """Expands a list of values based on non-null adjacent cells.
+
+    Expand can be done in 3 directions defined in :class:`~gspread.utils.TableDirection`
+
+        * ``TableDirection.right``: expands only same row as starting cell on the right side up to
+          first null/empty cell
+        * ``TableDirection.down``: expands only same column as starting cell to the bottom up to
+          first null/empty cell
+        * ``TableDirection.table``: expands in both direction, first right then down.
+
+    Regardless of the direction this function always returns a matrix of data, even if it has
+    only one column.
+
+    Example::
+
+        values = [
+            ['', '',   '',   '', ''  ],
+            ['', 'B2', 'C2', '', 'E2'],
+            ['', 'B3', 'C3', '', 'E3'],
+            ['', ''  , ''  , '', 'E4'],
+        ]
+        >>> utils.find_table(TableDirection.table, 'B2')
+        [
+            ['B2', 'C2'],
+            ['B3', 'C3'],
+        ]
+
+
+    .. note::
+
+       the ``TableDirection.table`` will first look right, then look down.
+       Any empty value in the midle of the table will be ignored.
```

Comment:
true, I will rework the docstring !

here in your example: it will stop on the first empty cell on the right, so V2 is the rightmost cell. Then it will go down until D2